nomad/grapl-core.nomad

```diff
@@ -0,0 +1,428 @@
+}
+
+# Where is redis located?
+variable "redis_endpoint" {
+    type = string
+}
+
+# What is the name of the schema table?
+variable "schema_table_name" {
+    type = string
+}
+
+# https://github.com/grapl-security/grapl/blob/af6f2c197d52e9941047aab813c30d2cbfd54523/pulumi/infra/dynamodb.py#L118
+# Without deployment name
+variable "session_table" {
+    type = string
+    default = "dynamic_session_table"
+}
+
+# How many graph mergers should be running in tandem
+variable "graph_mergers" {
+    type = number
+    default = 1
+}
+
+# How many node identifiers should be running in tandem
+variable "node_identifiers" {
+    type = number
+    default = 1
+}
+
+variable "node_identifiers_retry" {
+    type = number
+    default = 1
+}
+
+locals {
+    # DGraph Alphas (shards * replicas)
+    dgraph_alphas = [for alpha_id in range(0, var.dgraph_replicas * var.dgraph_shards): {
+        id: alpha_id,
+        grpc_private_port: alpha_id + 7080,
+        grpc_public_port: alpha_id + 9080,
+        http_port: alpha_id + 8080
+    }]
+
+    # DGraph Zeros (replicas)
+    dgraph_zeros = [for zero_id in range(1, var.dgraph_replicas): {
+        id: zero_id,
+        grpc_private_port: zero_id + 5080,
+    }]
+
+    # String that contains all of the Zeros for the Alphas to talk to and ensure they don't go down when one dies
+    zero_alpha_connect_str = join(",", [for zero_id in range(0, var.dgraph_replicas): "localhost:${zero_id + 5080}"])
+
+    # String that contains all of the running Alphas for clients connecting to Dgraph (so they can do loadbalancing)
+    alpha_grpc_connect_str = join(",", [for alpha in local.dgraph_alphas: "localhost:${alpha.grpc_public_port}"])
+}
+
+job "grapl-core" {
+    datacenters = ["dc1"]
+
+    type = "service"
+
+    # Specifies that this job is the most high priority job we have; nothing else should take precedence 
+    priority = 100
+
+    update {
+        # Automatically promotes to canaries if all canaries are healthy during an update / deployment 
+        auto_promote = true
+        # Auto reverts to the last stable job variant if the update fails 
+        auto_revert = true
+        # Spins up a "canary" instance of potentially destructive updates, validates that they are healthy, then promotes the instance to update
+        canary = 1
+        max_parallel = 1
+        # The min amount of reported "healthy" time before a instance is considered healthy and an allocation is opened up for further updates
+        min_healthy_time = "15s"
+    }
+
+    group "dgraph-zero-leader" {
+        network {
+            mode = "bridge"
+        }
+
+        task "dgraph-zero-leader" {
+            driver = "docker"
+
+            config {
+                image = "dgraph/dgraph:latest"
+                args = [
+                    "dgraph",
+                    "zero",
+                    "--my=localhost:5080",
+                    "--replicas", "${var.dgraph_replicas}",
+                    "--raft", "idx=1",
```

Comment:
Any significance to having the flag and value in a single value in one case (`--my=localhost:5080`), but split up in others (`--replicas`, `${var.dgraph_replicas}`)?

It would be nice to adopt a single convention.

Reply:
No particular point! I think I was doing a combination of copy+paste from dgraph docs and manually writing/messing about with the config to get it working originally.